pkg/apis/monitoring/v1/prometheus_types.go

```diff
@@ -56,6 +56,25 @@ func (l *Prometheus) GetStatus() PrometheusStatus {
 	return l.Status
 }
 
+// +kubebuilder:validation:Enum=ResourceName;ShardAndResourceName
+type AdditionalLabelSelector string
+
+const (
+	// Automatically add a label selector that will select all pods matching the same Prometheus/PrometheusAgent resource (irrespective of their shards).
+	ResourceNameLabelSelector AdditionalLabelSelector = "ResourceName"
+
+	// Automatically add a label selector that will select all pods matching the same shard.
+	ShardAndResourceNameLabelSelector AdditionalLabelSelector = "ShardAndResourceName"
+)
+
+type TopologySpreadConstraint struct {
+	*v1.TopologySpreadConstraint `json:",inline"`
+
+	//+optional
+	// Defines what Prometheus Operator managed labels should be added to labelSelector on the topologySpreadConstraint.
+	AdditionalLabelSelectors *AdditionalLabelSelector `json:"additionalLabelSelectors,omitempty"`
```

Comment:
I'm still wondering about the best naming for this field :thinking:

Reply:
Me too 😢 
But I ran out of ideas.

Reply:
ManagedLabelsToLabelSelector?

I don't think we'll find the perfect name here 😅 , `additionalLabelSelectors` is also fine for me

Reply:
I'm fine with both!
@simonpasquier any other feedback about this PR?